internal/rbd/replicationcontrollerserver.go

```diff
@@ -208,10 +208,22 @@ func (cs *ControllerServer) DisableVolumeReplication(ctx context.Context,
 			util.ErrorLog(ctx, err.Error())
 			return nil, status.Error(codes.Internal, err.Error())
 		}
-		// TODO: check do we need to return abort after disabling the mirroring
+		// the image state can be still disabling once we disable the mirroring
+		// check the mirroring is disabled or not
+		mirroringInfo, err = rbdVol.getImageMirroringInfo()
+		if err != nil {
+			util.ErrorLog(ctx, err.Error())
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		if mirroringInfo.State == librbd.MirrorImageDisabling {
+			return nil, status.Errorf(codes.Aborted, "%s is in disabling state", volumeID)
+		}
 		return &replication.DisableVolumeReplicationResponse{}, nil
+	// image is already in disabled state
 	case librbd.MirrorImageDisabled:
-		// image is already in disabled state
+	// image mirroring is still disabling
+	case librbd.MirrorImageDisabling:
```

Comment:
Please move `Disabled` and `Disabling` cases above `Enabled`. It will avoid extra operations if image is in any of the disabled/disabling state.